Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SimulatedDensityOfStates.py

```diff
@@ -333,11 +249,132 @@ def _get_properties(self):
         self._peak_width = self.getProperty('PeakWidth').value
         self._scale = self.getProperty('Scale').value
         self._zero_threshold = self.getProperty('ZeroThreshold').value
-        self._ions = self.getProperty('Ions').value
+        self._ions_of_interest = self.getProperty('Ions').value
         self._sum_contributions = self.getProperty('SumContributions').value
         self._scale_by_cross_section = self.getPropertyValue('ScaleByCrossSection')
-        self._calc_partial = (len(self._ions) > 0)
+        self._calc_partial = (len(self._ions_of_interest) > 0)
+
+
+#----------------------------------------------------------------------------------------
+
+    def _create_ion_table(self, unit_cell, ions):
+        """
+        Creates an ion table from the data in the phonon/castep file
+        """
```

Comment:
I think the phonon/castep data has already been read/loaded into self, in which case the comment should state this. E.g. "... data read from phonon/castep file".